ang/test/civicase/ActivitiesCalendar.spec.js

```diff
@@ -533,6 +544,54 @@
       });
     });
 
+    describe('when the date picker selects a month', function () {
+      var allArgs, endOfMonth, nextMonth, startOfMonth;
+
+      beforeEach(function () {
+        nextMonth = moment(dates.today).add(1, 'month');
+        startOfMonth = nextMonth.startOf('month').format('YYYY-MM-DD');
+        endOfMonth = nextMonth.endOf('month').format('YYYY-MM-DD');
+
+        initControllerAndEmitDatepickerReadyEvent();
+        crmApi.calls.reset();
+        $rootScope.$emit('civicase::uibDaypicker::monthSelected', nextMonth.toDate());
+        $scope.$digest();
+
+        allArgs = crmApi.calls.allArgs();
+      });
+
+      // @NOTE: the function that loads the data when a new month is selected
+      // is debounced to avoid flooding, and as such can't be tested normally
+      // The use of `done` and `setTimeout` is necessary for jasmine to
+      // wait the next tick loop before testing the assertions
+
+      it('does not invoke the load function before 300ms', function (done) {
+        setTimeout(function () {
+          expect(crmApi).not.toHaveBeenCalled();
+          done();
+        }, 290);
+      });
+
+      it('invokes the load function after 300ms', function (done) {
+        setTimeout(function () {
+          expect(crmApi).toHaveBeenCalled();
+          done();
+        }, 300);
+      });
+
+      it('loads the days with activities of the month of the selected date', function (done) {
+        setTimeout(function () {
+          expect(crmApi.calls.count()).toBe(2);
+          allArgs.forEach(function (args) {
+            expect(args[2].activity_date_time).toEqual({
+              'BETWEEN': [startOfMonth + ' 00:00:00', endOfMonth + ' 23:59:59']
+            });
+          });
+          done();
+        }, 300);
+      });
+    });
```

Comment:
we can use Jasmine clock's for these kind of tests:
https://jasmine.github.io/2.9/introduction#section-Jasmine_Clock

Not sure if it would be a better option. But it would be good to decide if we do these kind of tests using the jasmine clock, or timeouts.